packages/mobile/src/screens/upload-screen/screens/LicenseTypeScreen.tsx

```diff
@@ -0,0 +1,177 @@
+import { computeLicenseIcons } from '../utils/computeLicenseIcons'
+
+const messages = {
+  title: 'License Type',
+  allowAttribution: 'Allow Attribution',
+  noAttribution: 'No Attribution',
+  commercialUse: 'Commercial Use',
+  nonCommercialUse: 'Non-Commercial Use',
+  derivativeWorks: 'Derivative Works',
+  allowed: 'Allowed',
+  shareAlike: 'Share-Alike',
+  notAllowed: 'Not Allowed'
+}
+
+const allowAttributionValues = [
+  { key: true, text: messages.allowAttribution },
+  { key: false, text: messages.noAttribution }
+]
+
+const commercialUseValues = [
+  { key: true, text: messages.commercialUse },
+  { key: false, text: messages.nonCommercialUse }
+]
+
+const derivativeWorksValues = [
+  { key: null, text: messages.allowed },
+  { key: true, text: messages.shareAlike },
+  { key: false, text: messages.notAllowed }
+]
+
+const useStyles = makeStyles(({ spacing, palette, typography }) => ({
+  content: {
+    paddingHorizontal: spacing(6),
+    paddingVertical: spacing(8)
+  },
+  licenseSection: {
+    marginBottom: spacing(6)
+  },
+  licenseSectionLabel: {
+    marginBottom: spacing(2)
+  },
+  divider: {
+    marginBottom: spacing(6)
+  },
+  licenseDescription: {
+    padding: spacing(4),
+    backgroundColor: palette.neutralLight10,
+    borderWidth: 1,
+    borderColor: palette.neutralLight8,
+    borderRadius: spacing(2)
+  },
+  licenseIcons: {
+    flexDirection: 'row',
+    marginBottom: spacing(2)
+  },
+  licenseIcon: {
+    marginRight: spacing(1)
+  },
+  descriptionText: {
+    lineHeight: typography.fontSize.small * 1.5
+  }
+}))
+
+export const LicenseTypeScreen = () => {
+  const [{ onChange }] = useField('license')
+  const [{ value: allowAttribution }, , { setValue: setAllowAttribution }] =
+    useField<boolean>('licenseType.allowAttribution')
+  const [{ value: commercialUse }, , { setValue: setCommercialUse }] =
+    useField<boolean>('licenseType.commercialUse')
+  const [{ value: derivativeWorks }, , { setValue: setDerivateWorks }] =
+    useField<Nullable<boolean>>('licenseType.derivativeWorks')
+
+  const styles = useStyles()
+  const { neutral } = useThemeColors()
+
+  const { licenseType, licenseDescription } = computeLicense(
+    allowAttribution,
+    commercialUse,
+    derivativeWorks
+  )
+
+  const licenseIcons = computeLicenseIcons(
+    allowAttribution,
+    commercialUse,
+    derivativeWorks
+  )
+
+  useEffect(() => {
+    onChange('license')(licenseType)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [licenseType])
+
+  const labelProps: TextProps = {
+    fontSize: 'large',
+    weight: 'demiBold',
+    style: styles.licenseSectionLabel
+  }
+
+  return (
+    <UploadStackScreen
+      title={messages.title}
+      icon={IconCreativeCommons}
+      variant='white'
+    >
+      <View style={styles.content}>
+        <View style={styles.licenseSection}>
+          <Text {...labelProps}>{messages.allowAttribution}?</Text>
```

Comment:
Is there a reason to keep the `?` outside `messages`? That might not be appropriate for all localizations

Reply:
Great callout, I was using messages in multiple spots so I should probably just duplicate them